### PR TITLE
virttest.qemu_qtree: Update guess_type() for QtreeDev

### PR DESCRIPTION
**Pull request by spcui**
Make the get method can get the correct QtreeDisk device
on different platform.

ID: 1211151 1231011

### CODE REVIEW

**Comment by CongSmile** [comment]
Hi @spcui 

The request is that virtio-blk-pci info in qtree is different in RHEL.6 and RHEL.7, we need update the code to the root cause I think.
I don't think that disabling the virtio-blk-pci can solve the problem.

Please be free to correct me if I'm misunderstanding.

Thanks.

**Comment by spcui** [comment]
Hi @CongSmile 

For rhel7 host, self.qtree['type'] == 'virtio-blk-device', and return disk, 
for rhel6 host, self.qtree['type'] == 'virtio-blk-pci', and elif branch can return the correct type.

**Comment by CongSmile** [comment]
Thx for the explanation at first.

For the elif branch, you don't judge "virtio-blk-pci", right?
 which means that no matter qtree['type'] equals to "virtio-blk-pci", it will return the QtreeDisk.

But actually, it qtree['type'] != "virtio-blk-pci", it should return QtreeDev, right?

Please be free to correct me if I'm misunderstanding.

Thanks.

**Comment by spcui** [comment]
HI @CongSmile updated, could you help review again?

**Comment by CongSmile** [comment]
Acked-by: Cong Li <coli@redhat.com>

**Comment by xutian** [comment]
@spcui 

for guess_type in Qdev,  it just get dev objects in qtree, for guess_type in qdisk, it should be overwrite in Qdisk not in Qdev.
so Nack this PR.

Thanks,
Xu

**Comment by spcui** [comment]
@xutian @CongSmile updated, could you review again?

**Comment by CongSmile** [comment]
Hi @spcui 

CI check failed, could you fix it at first?

Thx.

File: virttest/qemu_qtree.py [hunk 1]
```diff
@@ -471,12 +479,14 @@ def check_guests_proc_scsi(self, info):
             else:
                 proc_not_scsi += 1
         for disk in disks.difference(scsis):
-            logging.error('Disk %s is in qtree but not in /proc/scsi/scsi.',
-                          disk)
+            error_msg = 'Disk %s is in qtree but not in /proc/scsi/scsi' % disk
+            logging.error(error_msg)
```
Comment (vipmike007) [inline virttest/qemu_qtree.py]: Does this function work for windows?
If yes looks the error_msg is not correct.

File: virttest/qemu_qtree.py [hunk 2]
```diff
@@ -471,12 +479,14 @@ def check_guests_proc_scsi(self, info):
...
+            self.errors.append(error_msg)
             additional += 1
         for disk in scsis.difference(disks):
-            logging.error('Disk %s is in /proc/scsi/scsi but not in qtree.',
-                          disk)
+            error_msg = 'Disk %s is in /proc/scsi/scsi but not in qtree' % disk
+            logging.error(error_msg)
```
Comment (vipmike007) [inline virttest/qemu_qtree.py]: same as above

**Comment by vipmike007** [comment]
inspekt style
/home/travis/build/autotest/virt-test/virttest/qemu_qtree.py:364:5: E125 20
PEP8 check fail: /home/travis/build/autotest/virt-test/virttest/qemu_qtree.py
Trying to fix errors with autopep8
PEP8 compliance FAIL
The command "inspekt style" exited with 1.

File: virttest/qemu_qtree.py [hunk 3]
```diff
@@ -163,15 +163,7 @@ def add_child(self, child):
...
-        if self.qtree['type'] == 'virtio-blk-device':
-            return QtreeDisk
-        elif ('drive' in self.qtree and
-              self.qtree['type'] != 'usb-storage' and
-              self.qtree['type'] != 'virtio-blk-pci'):
-            # ^^ HOOK when usb-storage-containter is detected as disk
-            return QtreeDisk
-        else:
-            return QtreeDev
+        return QtreeDev
```
Comment (ypu) [inline virttest/qemu_qtree.py]: Your update will make the guess_type meaningless.

Can you separated your pullreq to two? one is for the debug log and another is for guess_type. We can make the debug log one in tree first.

**Comment by lmr** [comment]
Hi @spcui. As you might know, we're moving virt-test to avocado-vt, so I please ask you to consider re-implementing this functionality in the later. The new repo is:

https://github.com/avocado-framework/avocado-vt

Please keep in mind the following guidelines when creating the new PR:

1) Most of the time you can pick the patches of your original PR and merge them on a new branch with minimal changes.
3) Don't use autotest APIs, since we're moving away from using them. Please review the code and change things like `utils.system` -> `process.system`, `utils.run` -> `process.run`, so on and so forth.

Thanks, and I'm sorry for any inconveniences we might have caused your team.

**Pull request closed without merging**